shazam/shazam.py

```
@@ -0,0 +1,174 @@
+                    image_data = await response.read()
+                    color_thief = ColorThief(io.BytesIO(image_data))
+                    dominant_color = color_thief.get_color(quality=1)
+                    return discord.Color.from_rgb(*dominant_color)
+        except Exception as e:
+            logging.exception("Error fetching dominant color from image: %s", image_url, exc_info=e)
+            raise RuntimeError("Failed to fetch dominant color from the image.") from e
+
+    async def download_and_convert_to_audio(self, video_url: str) -> str:
+        """Download the video and convert it to an audio file, returning the URL of the uploaded audio."""
+        async with aiohttp.ClientSession() as session:
+            async with session.get(video_url) as response:
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)